0.preprocess-sites/7.visualize-cell-summary.py

```diff
@@ -0,0 +1,249 @@
+
+# Read and Merge Data
+cell_quality_list = []
+metadata_list = []
+metadata_col_list = []
+metadata_col_list.append("Cell_Class")
+metadata_col_list.extend(cell_cols)
+metadata_col_list.extend(barcode_cols)
+metadata_col_list.extend(gene_cols)
+metadata_col_list.extend(spot_score_count_cols)
+metadata_col_list.extend(spot_score_mean_cols)
+metadata_col_list.append(cell_quality_col)
+metadata_col_list.append(foci_site_col)
+
+input_dir = pathlib.Path(input_basedir, batch, "paint")
+sites = os.listdir(input_dir)
+print(f"There are {len(sites)} sites.")
+
+for site in sites:
+
+    cell_count_file = pathlib.Path(input_dir, site, f"cell_counts_{site}.tsv")
+    metadata_file = pathlib.Path(input_dir, site, f"metadata_{site}.tsv.gz")
+
+    # Aggregates cell quality by site into single list
+    cell_quality_list.append(pd.read_csv(cell_count_file, sep='\t'))
+
+    # Aggregates metadata by site into a single list
+    metadata_df = (
+        pd
+        .read_csv(metadata_file, sep='\t')
+        .loc[:,metadata_col_list]
+        .reset_index(drop=True)
+    )
+
+    metadata_list.append(metadata_df)
+
+# Creates dataframe from cell quality list
+cell_count_df = pd.concat(cell_quality_list, axis="rows").reset_index(drop=True)
+
+# Assigns the Cell_Quality column to the category datatype and sets categories
+cell_count_df.loc[:, 'Cell_Quality'] = (
+    pd.Categorical(
+        cell_count_df.Cell_Quality,
+        categories=cell_category_order
+    )
+)
+
+# Assigns the Site column to the category datatype
+cell_count_df.loc[:, "site"] = (
+    pd.Categorical(
+        cell_count_df.site,
+        categories=(
+            cell_count_df
+            .groupby("site")
+            ["cell_count"]
+            .sum()
+            .sort_values(ascending=False)
+            .index
+            .tolist()
+        )
+    )
+)
+
+cell_count_df = cell_count_df.assign(
+    Plate=[x[0] for x in cell_count_df.site.str.split("-")],
+    Well=[x[1] for x in cell_count_df.site.str.split("-")],
+    Site=[x[2] for x in cell_count_df.site.str.split("-")]
+)
+
+output_file = pathlib.Path(output_resultsdir, "cells", "cell_count.tsv")
+#cell_count_df.to_csv(output_file, sep='\t', index=False)
+
+print(cell_count_df.shape)
+cell_count_df.head(10)
+
+# Graph: Cell count with all wells in same graph
+cell_count_gg = (
+    gg.ggplot(cell_count_df, gg.aes(x="Site", y="cell_count"))
+    + gg.geom_bar(gg.aes(fill="Cell_Quality"), stat="identity")
+    + gg.theme_bw()
+    + gg.theme(axis_text_x = gg.element_text(rotation=90, size=5))
+    + gg.xlab("Sites")
+    + gg.ylab("Cell Count")
+    + gg.scale_fill_manual(
+        name="Cell Quality",
+        labels = cell_category_list,
+        values = cell_category_colors
+    )
+)
+
+output_file = pathlib.Path(output_figuresdir, "all_cellpainting_cellquality_across_sites.png")
+#cell_count_gg.save(output_file, dpi=300, width=10, height=7, verbose=False)
+
+cell_count_gg
+
+#Same graph as above, separated by Well.
```

Comment:
This graph isn't separating by well as desired.
Also, the row and ncol will need to be calculated instead of manually set

Reply:
`well_order` isn't probably encoded in the `site` column, correct? There should be some other column that has well information. After fixing this and the graph still doesn't work, lets setup a time to chat (today if possible)